ouroboros-network/src/Ouroboros/Network/Diffusion/Common.hs

```diff
@@ -193,14 +253,514 @@ data Applications ntnAddr ntnVersion ntnVersionData
+           PeerSelectionGovernorArgs extraState extraDebugState extraActions extraFlags extraPeers
+                                     extraAPI extraCounters
+                                     peeraddr (PeerConnectionHandle
+                                                 muxMode responderCtx peeraddr
+                                                 ntnVersionData bytes m a b)
+                                     exception m
+
+      -- | Function that computes extraCounters from PeerSelectionState
+      --
+    , daPeerSelectionStateToExtraCounters
+        :: forall muxMode responderCtx ntnVersionData bytes a b .
+           PeerSelectionState extraState extraFlags extraPeers
+                              peeraddr (PeerConnectionHandle
+                                          muxMode responderCtx peeraddr
+                                          ntnVersionData bytes m a b)
+        -> extraCounters
+
+      -- | Function that constructs a 'extraPeers' set from a map of dns
+      -- lookup results.
+      --
+    , daToExtraPeers :: Map peeraddr PeerAdvertise -> extraPeers
 
-      -- | Callback provided by consensus to inform it if the node is
-      -- connected to only local roots or also some external peers.
+      -- | Request Public Root Peers.
       --
-      -- This is useful in order for the Bootstrap State Machine to
-      -- simply refuse to transition from TooOld to YoungEnough while
-      -- it only has local peers.
+      -- If no custom public root peers is provided (i.e. Nothing) just the
+      -- default one from
+      -- 'Ouroboros.Network.PeerSelection.PeerSelectionActions.getPublicRootPeers'
       --
-    , daUpdateOutboundConnectionsState :: OutboundConnectionsState -> STM m ()
+    , daRequestPublicRootPeers
+        :: Maybe (PeerActionsDNS peeraddr resolver resolverError m
+        -> DNSSemaphore m
+        -> (Map peeraddr PeerAdvertise -> extraPeers)
+        -> ( (NumberOfPeers -> LedgerPeersKind -> m (Maybe (Set peeraddr, DiffTime)))
+        -> LedgerPeersKind
+        -> Int
+        -> m (PublicRootPeers extraPeers peeraddr, DiffTime)))
+
+      -- | Peer Churn Governor if no custom churn governor is required just
+      -- use the default one from
+      -- 'Ouroboros.Network.PeerSelection.Churn.peerChurnGovernor'
+      --
+    , daPeerChurnGovernor
+        :: PeerChurnArgs
+             m
+             extraChurnArgs
+             extraDebugState
+             extraFlags
+             extraPeers
+             extraAPI
+             extraCounters
+             peeraddr
+        -> m Void
+
+      -- | Provide extraActions to be passed to peer selection actions
+      --
+    , daExtraActions :: extraActions
+
+      -- | Provide extraChurnArgs to be passed to churn governor
+      --
+    , daExtraChurnArgs :: extraChurnArgs
+
+      -- | Extension point for third party users to be able to add more
+      -- arguments.
+    , daExtraArgs    :: extraArgs
```

Comment:
Is this used anywhere?  I don't see how it is used inside `Ouroboros.Network.Diffusion.P2P.run` function. This indicates that it could be removed.